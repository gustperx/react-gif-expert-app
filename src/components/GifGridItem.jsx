import PropTypes from "prop-types";

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="animate__animated animate__bounceIn">
      <img className="rounded-lg object-cover h-40" src={url} alt={title} />
      <p className="text-center mt-4 text-sm font-sans font-medium text-cyan-600">
        {title}
      </p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
