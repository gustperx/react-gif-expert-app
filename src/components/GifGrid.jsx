import PropTypes from "prop-types";
import { useFetchGifs } from "../hooks/useFetchGifs";

import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data, loaging } = useFetchGifs(category);

  return (
    <>
      <h3 className="text-2xl text-cyan-700 font-medium mb-4 animate__animated animate__fadeInDown">
        {category}
      </h3>

      {loaging && (
        <p className="text-2xl text-red-500 my-2 text-center font-semibold animate__animated animate__fadeInLeft">
          Loading...
        </p>
      )}

      <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-5">
        {data.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>

      <hr className="my-8 font-semibold" />
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
