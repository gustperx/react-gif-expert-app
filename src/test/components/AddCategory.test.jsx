import { shallow } from "enzyme";
import { afterEach, beforeEach, describe, expect, test, vi } from "vitest";
import { AddCategory } from "../../components/AddCategory";

describe("Pruebas en <AddCategory />", () => {
  const setCategories = vi.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test("debe de hacer match con el snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de cambiar la caja de texto", () => {
    const input = wrapper.find("input");
    const value = "Hola Mundo";
    input.simulate("change", { target: { value } });
  });

  test("NO debe de postear la informacion con submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test("debe de llamar el setCategories y limpiar la caja de texto", () => {
    // 1. simular inputChange
    const value = "Hola Mundo";
    wrapper.find("input").simulate("change", { target: { value } });

    // 2. Simular submit
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    // 3. setCategories se debe haber llamanado
    //expect(setCategories).toHaveBeenCalled();
    //expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

    // 4. el valor del input debe ser ''
    expect(wrapper.find("input").prop("value")).toBe("");
  });
});
