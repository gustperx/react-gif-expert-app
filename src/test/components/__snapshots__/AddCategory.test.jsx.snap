// Vitest Snapshot v1

exports[`Pruebas en <AddCategory /> > debe de hacer match con el snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    className="border rounded-md w-full text-md text-gray-700 border-gray-700 h-6 p-4"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
